client/proxy/proxy.go

```diff
@@ -21,6 +21,15 @@ type ManagerClient struct {
 	client           *http.Client
 }
 
+// Client defines the interface for the proxy specific functionality of the manager client
+type Client interface {
+	GetConfig(model.Key) (*apiserver.Config, error)
```

Comment:
I'm confused how come ParsedSpec is transferred over the wire if it's marked as `json:"-"`.